packages/autorest.go/src/generator/constants.ts

```diff
@@ -33,10 +23,12 @@ export async function generateConstants(session: Session<CodeModel>): Promise<st
   }
   // data-plane clients must manage their own constants for these values
   if (<boolean>session.model.language.go!.azureARM) {
-    const version = await getModuleVersion(session);
+    if (session.model.language.go!.moduleVersion === '') {
+      throw new Error('--module-version is a required parameter');
+    }
```

Comment:
You don't need this bc it has been handled in `namer.ts`.

Reply:
That was a bug.  `module-version` is not required by default but is required when `--azure-arm` is set.  I will fix the bug and add this back with a better error message.